### DISCUSSION TITLE
use environment variable in json

### DISCUSSION THREAD

**Discussion by mikethecalamity** [renovatebot/renovate#29493] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

How do I access an environment variable in my `renovate.json`? In my base project I have a `config.js` with `process.env.MY_REGISTRY`, that works. But in my discovered project, how would I get this variable? I can't seem to find any documentation on this. I tried: `registryUrlTemplate: "${{MY_REGISTRY}}"`, which I thought I saw in some documentation, but it didn't work.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Repo config does not automatically get access to env by default, for security reasons.

You can set exposeAllEnv=true in config.js and then use `"{{ env.MY_REGISTRY }}"` in renovate.json.

**Comment by mikethecalamity** [comment]
That worked, thanks!
